tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts

```diff
@@ -2576,29 +2613,58 @@ describe('scalar card', () => {
         });
         const fixture = createComponent('card1');
         fixture.detectChanges();
-        const testController = fixture.debugElement.query(
+        let testController = fixture.debugElement.query(
           By.directive(CardFobControllerComponent)
         ).componentInstance;
         const controllerStartPosition =
           testController.root.nativeElement.getBoundingClientRect().left;
 
-        // Simulate dragging start fob to step 25
-        testController.startDrag(Fob.START);
+        // Simulates dragging start fob to step 25
+        testController.startDrag(Fob.START, TimeSelectionAffordance.FOB);
         let fakeEvent = new MouseEvent('mousemove', {
           clientX: 25 + controllerStartPosition,
           movementX: 1,
         });
         testController.mouseMove(fakeEvent);
+        testController.stopDrag();
+        fixture.detectChanges();
+
+        expect(dispatchedActions).toEqual([
```

Comment:
Given that you test the dispatchedActions in 'dispatches linkedTimeSelectionChanged action when fob is dragged' and this test is not at all about linkedtime, I would be fine removing this check. The scope of the test is getting pretty confusing.  I'm not sure if any of the other changes you made in the test are necessary or not.

Reply:
The flow of this test is 
(1)drag start step to 25 --> (2)drag end step to 28 --> (3)disable linked time --> (4)check the fob is at step 25
This means the update of step selector is only reflected when updating linked time start step

However from (1) to (2), in this test it won't update the store state. I need to override selector (the paragraph of "Reflects linked time selection changes above."). Without checking `dispatchedActions` here, it makes no difference to me if (1) exists or not. I could just start with override selector in line 2649.

Does this make sense? Or let me know if you have any suggestions.

Reply:
I have no suggestions. Whatever makes most sense to you.